backend/src/api/monthReports/monthReport.controller.js

```diff
@@ -0,0 +1,57 @@
+const catchAsync = require('../../utils/catchAsync');
+const MonthReportModel = require('../cron/monthReport.model');
+const AppError = require('../errors/appError');
+
+const getMonthReport = catchAsync(async (req, res, next) => {
+  let { startMonth, startYear } = req.query;
+
+  if (Object.keys(req.query).length === 0) {
+    return next(new AppError('Required query params are not provided', 400));
+  }
+  if (!req.user) {
+    return next(new AppError('Not authorized', 401));
+  }
+  if (!req.user.flatPrice) {
+    return next(new AppError('Initialize your saving stats, please', 403));
+  }
+
+  let endDate;
+  let monthReports;
+  const userId = req.user._id;
+
+  if (startMonth === '' && startYear === '') {
```

Comment:
Second startMonth and startYear should have some values. If you want to have some default values for them on backend - they should not be present at all. `?startMonth=&startYear=` - this request should not exist at all. Third, if you need default values for some parameters - again, it's `joi` responsibility